src/main/java/software/amazon/awssdk/crt/auth/credentials/X509CredentialsProvider.java

```diff
@@ -48,6 +48,21 @@ public X509CredentialsProviderBuilder withClientBootstrap(ClientBootstrap client
             return this;
         }
 
+        /**
```

Comment:
I think it would be better to remove this and have the null bootstrap no longer be an error in the provider constructor and instead use the default there.